numpy/_array_api/_array_object.py

```diff
@@ -0,0 +1,891 @@
+"""
+Wrapper class around the ndarray object for the array API standard.
+
+The array API standard defines some behaviors differently than ndarray, in
+particular, type promotion rules are different (the standard has no
+value-based casting). The standard also specifies a more limited subset of
+array methods and functionalities than are implemented on ndarray. Since the
+goal of the array_api namespace is to be a minimal implementation of the array
+API standard, we need to define a separate wrapper class for the array_api
+namespace.
+
+The standard compliant class is only a wrapper class. It is *not* a subclass
+of ndarray.
+"""
+
+from __future__ import annotations
+
+import operator
+from enum import IntEnum
+from ._creation_functions import asarray
+from ._dtypes import _boolean_dtypes, _integer_dtypes, _floating_dtypes
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._types import Optional, PyCapsule, Tuple, Union, array
+
+import numpy as np
+
+class ndarray:
```

Comment:
From what I understand this class attempts to represent an interface of the abstract [array object](https://data-apis.org/array-api/latest/API_specification/array_object.html), and not specifically `ndarray`. 
If so, would something along the lines of `AbstractArray` not be a better suited name here?

Reply:
Makes sense. Maybe `Array`, to just use proper PEP 8 capitalization for a class, and avoid the same name that `np.array` uses?